### PR TITLE
Update native contract demos

### PR DESCRIPTION
**Pull request by shelvenzhou**

### CODE REVIEW

File: crates/phactory/src/contracts/btc_price_bot.rs [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+    }
+
+    /// Handle the Commands from transactions on the blockchain. This method doesn't respond.
+    fn handle_command(
+        &mut self,
+        origin: MessageOrigin,
+        cmd: Command,
+        context: &mut NativeContext,
+    ) -> TransactionResult {
+        info!("Command received: {:?}", &cmd);
```
Comment (h4x3rotab) [inline crates/phactory/src/contracts/btc_price_bot.rs:122]: Make it a comment. If we do this, the secrets in the command will be leaked to the miners

File: crates/phactory/src/contracts/btc_price_bot.rs [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+                        let text = format!("BTC price: ${}", price.usd);
+                        let uri = format!(
+                            "https://api.telegram.org/bot{}/{}",
+                            bot_token, "sendMessage"
+                        );
+                        let data = &TgMessage { chat_id, text };
+
+                        let mut resp = match surf::post(uri)
+                            .body_json(data)
+                            .expect("should not fail with valid data; qed.")
```
Comment (h4x3rotab) [inline crates/phactory/src/contracts/btc_price_bot.rs:207]: What will happen if the result is not a json? Will the worker crash as a whole?

File: crates/phactory/src/contracts/btc_price_bot.rs (hunk 1 above)
Comment (shelvenzhou) [reply crates/phactory/src/contracts/btc_price_bot.rs]: done

File: crates/phactory/src/contracts/btc_price_bot.rs (hunk 2 above)
Comment (shelvenzhou) [reply crates/phactory/src/contracts/btc_price_bot.rs]: return an error instead of panic

**Pull request merged**